models/icepack/model_mod.f90

```diff
@@ -0,0 +1,885 @@
+      print*,'obstype SIT expected conc:',expected_aggr_conc
+      print*,'SIT sic lat lon:',llat,llon
+      endif
+
+endif
+
+if (debug > 1) print *, 'interp val, istatus = ', expected_obs, istatus, size(expected_obs)
+
+! This should be the result of the interpolation of a
+! given obs_type of variable at the given location.
+
+! The return code for successful return should be 0. 
+! Any positive number is an error.
+! Negative values are reserved for use by the DART framework.
+
+end subroutine model_interpolate
+
+!------------------------------------------------------------------
+
+subroutine lon_lat_interpolate(state_handle, ens_size, offset, lon, lat, var_type, cat_signal, expected_obs, istatus)
+type(ensemble_type), intent(in)  :: state_handle
+integer,             intent(in)  :: ens_size
+integer(i8),         intent(in)  :: offset
+real(r8),            intent(in)  :: lon, lat
+integer,             intent(in)  :: var_type
+integer,             intent(in)  :: cat_signal
+real(r8),            intent(out) :: expected_obs(ens_size)
+integer,             intent(out) :: istatus(ens_size)
+
+real(r8) :: work_expected_obs(ens_size)
+integer  :: e, iterations, Niterations
+integer(i8) :: state_index
+
+if ( .not. module_initialized ) call static_init_model
+
+istatus = 0
+if (var_type == 14) then
+  e = 1
+else if (var_type == 15) then
+  e = 2
+else if (var_type == 16) then
+  e = 3
+endif
+if ( cat_signal < 1 )  then
+   Niterations = Ncat ! only iterate if aggregating over all types
+else
+   Niterations = 1 ! no need to iterate
+endif
+work_expected_obs = 0.0_r8
+expected_obs = 0.0_r8
+do iterations = 1, Niterations
+
+   ! FIXME: this should use the state structure routine 'get_dart_vector_index'
+   ! to get the start of the next category layer.  this code assumes it knows
+   ! exactly how the state vector is laid out (reasonable, but might not be true
+   ! in future versions of dart.)
+   !next_offset = offset + (iterations-1)*Nx 
+   !print*,'offset',offset
+   state_index = get_dart_vector_index(grid_oi,int(offset,i4),1, domain_id, e)
+   work_expected_obs = get_state(state_index,state_handle)
+   expected_obs = expected_obs+work_expected_obs
+enddo
+end subroutine lon_lat_interpolate
+
+!------------------------------------------------------------------
+! Given an integer index into the state vector structure, returns 
+! the associated location.
+
+subroutine get_state_meta_data(index_in, location, var_type)
+
+integer(i8),         intent(in)  :: index_in
+type(location_type), intent(out) :: location
+integer,             intent(out), optional :: var_type
+
+real(r8) :: lat, lon, rcat
+integer  :: ni_index, hold_index, cat_index, local_var, var_id
+! these should be set to the actual location and state quantity
+
+if ( .not. module_initialized ) call static_init_model
+
+call get_model_variable_indices(index_in, ni_index, cat_index, hold_index, var_id=var_id)
```

Comment:
get_model_variable_indices can return the qty so no need to also call 'get_state_kind'